### PR TITLE
accessibility improvement: native support web `aria-*` and `role`

### PR DESCRIPTION
**Pull request by TheEhsanSarshar**
as most of the core Tamagui components are using `aria-*` and `role` for accessibility, it means that accessibility is mostly broken in native, in this PR it's mapping those web-specific aria and role props to its React Native counterpart.

### CODE REVIEW

**Comment by natew** [comment]
Wonderful stuff, could you refactor it so the web won't load it? Basically put it into it's own file that's imported where `.native.ts` includes it, but `.web.ts` (or non-prefixed) doesn't. If it saves some runtime cost you can also use `if (process.env.TAMAGUI_TARGET === 'native')`

**Comment by TheEhsanSarshar** [comment]
done!

File: packages/web/src/helpers/getSplitStyles.tsx [hunk 1]
```diff
@@ -243,7 +249,49 @@ export const getSplitStyles: StyleSplitter = (
       if (keyInit === 'userSelect') {
         keyInit = 'selectable'
         valInit = valInit === 'none' ? false : true
-      } else if (keyInit.startsWith('data-') || keyInit.startsWith('aria-')) {
+      } else if (keyInit.startsWith('aria-*') || keyInit === 'role') {
+        if (webToNativeAccessibilityDirectMap[keyInit]) {
```
Comment (natew) [inline packages/web/src/helpers/getSplitStyles.tsx:253]: if process.env here to avoid this loading on web

Comment (natew) [reply packages/web/src/helpers/getSplitStyles.tsx:253]: i think this would error on web with null webToNativeAccessibilityDirectMap right?

**Comment by nderscore** [comment]
Doesn't React Native 0.71 natively support `role` and `aria-*` attributes already? 🤔 

https://reactnative.dev/blog/2023/01/12/version-071#web-inspired-props-for-accessibility-styles-and-events

**Comment by TheEhsanSarshar** [comment]
@nderscore yeah version > 0.71.1 does support it. values are not fully compatible with each other tho. for example if you pass slider instead of adjustable it will crash

**Pull request merged**